From b573649e9059673e01f8d6611f53f2a5291d6e7a Mon Sep 17 00:00:00 2001
From: Jonathan Baecker <jonbae77@gmail.com>
Date: Fri, 27 Sep 2024 07:22:45 +0200
Subject: [PATCH v2] Respect `omit_endlist` flag in subtitle playlists

Ensure that when the `-hls_flags omit_endlist` option is set,
the `#EXT-X-ENDLIST` tag is also omitted from the `stream_vtt.m3u8`
subtitle playlist. This maintains consistency with the behavior
in other playlists when `omit_endlist` is specified.
---
 libavformat/hlsenc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/hlsenc.c b/libavformat/hlsenc.c
index 1e932b7..528ba0f 100644
--- a/libavformat/hlsenc.c
+++ b/libavformat/hlsenc.c
@@ -1676,7 +1676,7 @@ static int hls_window(AVFormatContext *s, int last, VariantStream *vs)
             }
         }

-        if (last)
+        if (last && (hls->flags & HLS_OMIT_ENDLIST)==0)
             ff_hls_write_end_list(hls->sub_m3u8_out);

     }
--
2.46.1

